### ISSUE TITLE
Switch to correct view if parent block is selected

### ISSUE THREAD

**Issue by nielslange** [woocommerce/woocommerce-gutenberg-products-block#5427]
## Describe the bug

When adding the cart block to a page and opening the list view, the view between the `Filled Cart` and the `Empty Cart` can only be switched by clicking on an inner block within these blocks, e.g. the `Cart Items block` or the `Image block`. When clicking on the parent blocks itself (`Filled Cart` or `Empty Cart`) within the list view, the view does not change.

## To reproduce

1. Create a test page.
2. Add the cart block.
3. Open the list view.
4. Click alternating and `Filled Cart` and `Empty Cart`.
5. See that the view stays on the initial view and does not change.

## Expected behavior

The view should not only change when clicking on one of the inner blocks, but also when clicking on one of the parent blocks.

## Screenshots

<table>
<tr>
<td>Filled cart
<br><br>

![Filled cart](https://user-images.githubusercontent.com/3323310/146918638-e69500d8-c228-478b-905f-380a596da34c.png)
</td>
<td>Empty Cart
<br><br>

![Empty Cart](https://user-images.githubusercontent.com/3323310/146918652-3cdea846-4273-497c-80c8-63d30ef1fdaf.png)
</td>
</tr>
</table>

## Additional context

In #5358, we fixed the view switch for inner blocks. This issue aims to fix the view switch for the parent blocks as well.

**Comment by tarhi-saad** [comment]
Thank you for reporting the bug @nielslange.

I followed the described steps above to reproduce the bug, but the view did change after clicking on the parent blocks. I made the tests using the latest trunk (Version 6.8.0-dev) both locally and using [Pressable](https://pressable.com/) (i.e. Created a plugin package in ZIP format for this one).

## Screenshots

<table>
<tr>
<td>Filled cart
<br><br>

![Filled cart](https://user-images.githubusercontent.com/14235870/149021094-ab669a73-3965-45de-9511-acf6b91b952e.png)
</td>
<td>Empty Cart
<br><br>

![Empty Cart](https://user-images.githubusercontent.com/14235870/149020862-a31dd5ae-a41b-4a02-aeef-97e000657377.png)
</td>
</tr>
</table>

On the other hand, the current version that can be installed from the plugins menu (i.e. Version: 6.7.1) does indeed have this bug. The view changes only when clicking on the inner blocks. 

If you did use the latest trunk for your tests maybe testing on a fresh WordPress install can give us more insight.

**Comment by nielslange** [comment]
I still see this problem, @tarhi-saad. When selecting the inner blocks, the view switches as expected. However, when selecting the parent blocks `Filled Cart` or `Empty Cart`, the view does not switch:

![ezgif-7-7c962f9a0a](https://user-images.githubusercontent.com/3323310/149080947-b575e6a6-f420-43b2-b926-e17bb2bd0078.gif)

**Comment by tarhi-saad** [comment]
Hi @nielslange! I've dug deeper into this issue:

TL;DR: This bug was fixed in #5446. It was merged into the trunk branch 11 days ago. 

> In #5358, we fixed the view switch for inner blocks. This issue aims to fix the view switch for the parent blocks as well.

As to why the view switch for the parent blocks wasn't working in the PR you've provided (i.e. #5358):
The logic was to compare the parent name of the selected block in the list view, with the currently displayed view name (see line [#46](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5358/files#diff-cf68455270c0b24563bd7b8e214d8937cc410150400df15fdea1dafe72537ad1R46) in this PR). This means if the selected element is `Empty Cart`, then we will check the name of its parent (i.e. `cart`). And this will not trigger a change of the Cart view.

On the other hand, it was fixed in #5446 by comparing the selected block name with the currently displayed view name (see line [#51](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5446/files#diff-cf68455270c0b24563bd7b8e214d8937cc410150400df15fdea1dafe72537ad1R51)). This means if the element is `Empty Cart`, then it'll be compared with the current view name (Empty Cart / Filled Cart). And in case of the child element is selected then we are checking the parent block name instead (see lines [#72-75](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5446/files#diff-cf68455270c0b24563bd7b8e214d8937cc410150400df15fdea1dafe72537ad1R72-R76)).

This last fix (#5446) is only available in the last release (6.8.0). That's why the bug is still occurring on the public version of the Woo Block plugin (i.e. v6.7.1).

A simple test I've done to make sure that this last PR #5446 is indeed fixing this bug:

1. Checkout to the trunk branch or the release/6.8.0
2. Copy the code in the file: `assets/js/blocks/cart-checkout/shared/use-view-switcher.tsx`
3. Checkout to the previous release (e.g. 6.7.2 or 6.7.1)
4. Test if the bug is occurring on this release
5. Then replace the code in the file `assets/js/blocks/cart-checkout/shared/use-view-switcher.tsx` by the previously copied code
6. Test again and try changing the view. The bug should be fixed!

**Comment by nielslange** [comment]
Great investigation, @tarhi-saad! 👏 Thank you so much for investigate the cause of this problem. I can confirm that the problem is solved with 6.8.0, based on the PR you mentioned. 

As this issue is solved, I'll close this PR.
